mantidproject/mantid
# GSASIIRefineFitPeaks XMin and XMax parameters added

Parameters for specifying a range of x values in which to do refinement were added to `GSASIIRefineFitPeaks`.

**To test:**

**Make sure the GSASIIRefineFitPeaks system test passes locally - this is not currently run on the build server**

You'll need GSAS-II installed. Either install it using one of the scripts in `scripts/GSAS-II` or follow the instructions [here](https://subversion.xray.aps.anl.gov/trac/pyGSAS).

Sample data is at `\\olympic\Babylon5\Public\JoeRamsay\GSASIIRefineFitPeaks`.

Run the following script:
```
import os
path_to_gsas = "/path/to/directory/containing/GSASIIscriptable.py"

data_dir = r"\\olympic\Babylon5\Public\JoeRamsay\GSASIIRefineFitPeaks"
# or wherever you've put it

ws_path = os.path.join(data_dir, "focused_bank1_ENGINX00256663.nxs")
spectrum = Load(Filename=ws_path)

gsas_proj_file = "/somewhere/local/Test.gpx"
phase_file = os.path.join(data_dir, "FE_ALPHA.cif")
iparams_file = os.path.join(data_dir, "template_ENGINX_241391_236516_North_bank.prm"

GSASIIRefineFitPeaks(PathToGSASII=path_to_gsas, 
                     InputWorkspace=spectrum,
                     SaveGSASIIProjectFile=gsas_proj_file,
                     PhaseInfoFile=phase_file,
                     InstrumentFile=iparams_file,
                     RefinementMethod="Pawley refinement",
                     XMin=10000, XMax=40000),
```

Open up `Test.gpx` in GSAS-II and have a look at it - make sure that `TMin` and `TMax` are set correctly in the `Limits` tab, and that fitted data is only visible between 10000 and 40000 us. Try varying `XMin` and `XMax` to see if anything unexpected happens.

Fixes #21600 

**Release Notes** 
This was a feature of the old version of the algorithm, so doesn't need to be in the release notes

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
